style-sync/src/main/scala/commercialtools/stylesync/DFP.scala

```diff
@@ -0,0 +1,99 @@
+package commercialtools.stylesync
+
+import com.google.api.ads.common.lib.auth.OfflineCredentials
+import com.google.api.ads.common.lib.auth.OfflineCredentials.Api
+import com.google.api.ads.dfp.axis.factory.DfpServices
+import com.google.api.ads.dfp.axis.utils.v201708.StatementBuilder
+import com.google.api.ads.dfp.axis.v201708.{CreativeTemplate, CreativeTemplateServiceInterface, NativeStyle, NativeStyleServiceInterface, NativeStyleStatus}
+import com.google.api.ads.dfp.lib.client.DfpSession
+import com.google.api.client.auth.oauth2.Credential
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Success, Try}
+
+// this is largely based off the documentation here: https://developers.google.com/doubleclick-publishers/docs/native
+object DFP extends Logging {
+  lazy val creds: Credential = new OfflineCredentials.Builder()
```

Comment:
Do these need to be lazy?

Reply:
I guess not - I sort of copied this bit from `frontend`. It does mean that we don't bother building the credentials until they are used, though given how objects are initialised, this would be the case anyway.... I'll remove the laziness!

Reply:
I would default to avoiding lazy unless necessary generally.